### ISSUE TITLE
Toggle button focus is difficult to determine

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#36439]
- VSCode Version: Code 1.17.2 (b813d12980308015bcd2b3a2f6efa5c810c33ba5, 2017-10-16T13:57:00.652Z)
- OS Version: Linux x64 4.10.0-35-generic
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
terminal-tabs|Tyr|0.1.1

---

Active + not focused:

![image](https://user-images.githubusercontent.com/2193314/31680572-904cb30e-b329-11e7-99e0-febcda4ffe03.png)

Active + focused:

![image](https://user-images.githubusercontent.com/2193314/31680583-9d8e22c8-b329-11e7-9085-d1c934af33c8.png)

**Comment by sandy081** [comment]
@misolori This is same for all our check boxes for eg in Search and Find widgets. Any suggestions would be appreciated here.

**Comment by Tyriar** [comment]
This is totally a bug imo, we have the theme keys, we just don't have any contrast at all between active and focus in our themes:

<img width="245" alt="screen shot 2018-09-14 at 8 00 52 am" src="https://user-images.githubusercontent.com/2193314/45558284-6178a580-b7f4-11e8-9213-94b8d8882a09.png">

Currently we're using two things to show state changes in the buttons:

- Weight: focus, active or hover
- Border: focus and active

This is different to the icons on the right of the find widget (next/prev match, find in selection, close), note that only find in selection has an active state here:

- Background change: hover (except find in selection which is wrong, related https://github.com/Microsoft/vscode/issues/50825)
- Border: focus

---

It's only in these toggles that we use border to indicate active which is why we see this issue. I propose we streamline all these icons to work the same:

- Weight: focus, active, hover (this isn't really needed)
- Background: active, hover
- Border: focus

This would come with the perk (con?) of not drawing so much attention to these toggles. Alternatively we could use a darker blue on dark themes and a lighter blue on light themes to indicate active/hover.

**Comment by Tyriar** [comment]
+ @roblourens since this also applies to the search viewlet.

**Comment by roblourens** [comment]
We talked about this in  https://github.com/Microsoft/vscode/issues/48855, we decided to handle it with https://github.com/Microsoft/vscode/issues/8017. What do you mean by "weight" here? The foreground color?

I'm not sure about getting rid of the border, I think it should be obvious when these toggles are on or off.

**Comment by misolori** [comment]
This is also related to #52161 where the "active" state can be confusing to decipher.

**Comment by Tyriar** [comment]
> What do you mean by "weight" here? The foreground color?

Yes the foreground color.

> I'm not sure about getting rid of the border, I think it should be obvious when these toggles are on or off.

It's an accessibility issue though that focus and active look the same, and the border is what we use everywhere to indicate focus.

**Comment by misolori** [comment]
@Tyriar do you think something like this would work?

![image](https://user-images.githubusercontent.com/35271042/45564681-0ea7e980-b806-11e8-8415-0e247195c189.png)

**Comment by Tyriar** [comment]
@misolori that's fine for hover, but focus/active states are the main problem. Which one of these is focused?

<img width="65" alt="screen shot 2018-09-14 at 12 39 14 pm" src="https://user-images.githubusercontent.com/2193314/45571493-36ee1300-b81b-11e8-8938-2ab78db340c4.png">

**Comment by misolori** [comment]
@Tyriar my screenshot above has the various states (default, hover, focus, and active). And I totally agree that focus/active are a problem, my suggestion was to add a background for active to help differentiate it from the focus border:

![image](https://user-images.githubusercontent.com/35271042/45572132-48d0b580-b81d-11e8-9319-a0625ca0f100.png)

**Comment by misolori** [comment]
Here's a gif that shows the keyboard interactions:
![gif](https://user-images.githubusercontent.com/35271042/45573235-c77b2200-b820-11e8-84b7-cad122064177.gif)

**Comment by Tyriar** [comment]
@misolori doh, I didn't even notice the background on the bottom right, yeah that looks good. The duller blue would probably be better than the really bright one (and be more obvious when the focus isn't on).

**Comment by misolori** [comment]
Update status: during our UX sync today, we decided that we are going to try out using a background color for an active state and use the outline for only the focus state. 

We'll try and re-use the `focusBorder` color and change the opacity (dark: `#0E639C` @ 60% opacity, light: `#007ACC` @ 20% opacity`). This will be applied to all toggle states (find widget, find in selection, exclude settings, etc.). 

That will end up looking like this:

![image](https://user-images.githubusercontent.com/35271042/46768570-7e8e8000-cc9d-11e8-9e8e-73ccf09c9994.png)

**Comment by sandy081** [comment]
Looks good

**Comment by isidorn** [comment]
@misolori @Tyriar has this been improved with the new icons?

**Comment by roblourens** [comment]
No, nothing has changed about how they are focused/highlighted.

**Comment by isidorn** [comment]
There are too many people assigned here.
@misolori assigning to you. It would be great if we can tackle this some time in the future. Since we are changing icons, this might go well with it.
